pkg/cmd/server/start.go

```diff
@@ -794,33 +204,40 @@ func env(key string, defaultValue string) string {
 	return val
 }
 
-func expandImage(component, defaultValue string, latest bool) string {
-	template := defaultValue
-	if s, ok := imageComponentEnvExpander(component); ok {
-		template = s
-	}
-	value, err := variable.ExpandStrict(template, func(key string) (string, bool) {
-		switch key {
-		case "component":
-			return component, true
-		case "version":
-			if latest {
-				return "latest", true
-			}
-		}
-		return "", false
-	}, variable.Versions)
+func (cfg Config) RunEtcd() error {
+	etcdAddr, err := cfg.GetEtcdAddress()
 	if err != nil {
-		glog.Fatalf("Unable to find an image for %q due to an error processing the format: %v", err)
+		return err
+	}
+
+	etcdConfig := &etcd.Config{
+		BindAddr:     getHostOfHost(cfg.BindAddr.URL.Host),
```

Comment:
could keep using cfg.BindAddr.Host (which holds host separate from port), and drop the getHostOfHost() (or call cfg.GetEtcdBindAddr() which does the same, yadda, yadda, yadda)